File: JavaScriptCode/M_156_BinaryTreeUpsideDown.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {TreeNode}
 */
var upsideDownBinaryTree = function(root) {
    if (root === null) return root;
    let l = root.left, r = root.right;
    root.left = null;
    root.right = null;
    while (l !== null) {
        let tl = l.left, tr = l.right;
        l.left = r;
        l.right = root;
        root = l;
        l = tl;
        r = tr;
    }    
    return root;
};

File: CppCode/M_156_BinaryTreeUpsideDown.cpp
#include "lc_header.h"
using namespace std;

class Solution {
public:
    TreeNode* upsideDownBinaryTree(TreeNode* root) {
        if (!root) return root;
        TreeNode *l = root->left, *r = root->right;
        root->left = nullptr;
        root->right = nullptr;
        while (l) {
            TreeNode *tl = l->left, *tr = l->right;
            l->left = r;
            l->right = root;
            root = l;
            l = tl;
            r = tr;
        }
        return root;
    }
};